x-pack/plugins/observability_solution/entity_manager/server/lib/auth/privileges.ts

```diff
@@ -31,34 +101,37 @@ export const requiredRunTimePrivileges = {
   ],
 };
 
-export const requiredEnablementPrivileges = {
-  // any one of
-  cluster: ['manage_security', 'manage_api_key', 'manage_own_api_key'],
+const entityDefinitionDeletionPrivileges = {
+  cluster: ['manage_transform', 'manage_ingest_pipelines'],
+  application: [
+    {
+      application: 'kibana-.kibana',
+      privileges: ['saved_object:entity-definition/delete'],
+      resources: ['*'],
+    },
+  ],
 };
 
-export const canRunEntityDiscovery = async (client: ElasticsearchClient) => {
-  const { has_all_requested: hasAllRequested } = await client.security.hasPrivileges({
-    body: {
-      cluster: requiredRunTimePrivileges.cluster,
-      index: requiredRunTimePrivileges.index,
-      application: requiredRunTimePrivileges.application,
+const apiKeyCreationPrivileges = {
+  // any one of
+  cluster: ['manage_security', 'manage_api_key', 'manage_own_api_key'],
+  application: [
+    {
+      application: 'kibana-.kibana',
+      privileges: ['saved_object:entity-discovery-api-key/*'],
+      resources: ['*'],
     },
-  });
-
-  return hasAllRequested;
+  ],
 };
 
-export const canEnableEntityDiscovery = async (client: ElasticsearchClient) => {
-  const [canRun, { cluster: grantedClusterPrivileges }] = await Promise.all([
-    canRunEntityDiscovery(client),
-    client.security.hasPrivileges({
-      body: {
-        cluster: requiredEnablementPrivileges.cluster,
-      },
-    }),
-  ]);
-
-  return (
-    canRun && requiredEnablementPrivileges.cluster.some((k) => grantedClusterPrivileges[k] === true)
-  );
+const apiKeyDeletionPrivileges = {
+  // any one of
+  cluster: ['manage_security', 'manage_api_key'],
```

Comment:
I didn't include `manage_own_api_key` as this would only allow the creator of the key to delete it. We're going to document these permissions and including `manage_own_api_key` as a sufficient permissions could be wrong in some cases